cvat/apps/engine/serializers.py

```diff
@@ -1380,6 +1388,17 @@ def update(self, instance: models.Task, validated_data: dict[str, Any]) -> model
                     "Invalid size of 'honeypot_real_frames' array, "
                     f"expected {task_honeypot_frames_count}"
                 )
+        elif frame_selection_method == models.JobFrameSelectionMethod.RANDOM_UNIFORM:
+            # Reset active honeypots in the task to produce a uniform distribution in the end
+            task_frame_provider = TaskFrameProvider(instance)
+            task_abs_disabled_validation_frames = set(
+                task_frame_provider.get_abs_frame_number(v)
+                for v in validation_layout.disabled_frames
+            )
+
+            for db_image in instance.data.images.filter(is_placeholder=True).all():
+                if db_image.real_frame not in task_abs_disabled_validation_frames:
+                    db_image.real_frame = -1
```

Comment:
Does that mean that `JobValidationLayoutWriteSerializer.update` expects honeypot images with real_frame == -1 (e.g., when initializing all_task_frames)? Because right now, it seems to me this does not happen and is not used.

Reply:
No, we're counting frame uses when selecting new honeypots, so here we reset the existing uses. -1 is just a dummy value.